api/odl.py

```diff
@@ -125,7 +115,7 @@ class ODLAPIConfiguration(ConfigurationGrouping):
         label=_("Passphrase hint"),
         description=_("Hint displayed to the user when opening an LCP protected publication."),
         type=ConfigurationAttributeType.TEXT,
-        required=False,
+        required=True,
```

Comment:
For already configured servers we need to handle the case where this configuration setting isn't already set. As you mentioned in notion, this can be handled by a database migration. I don't like the idea of adding a database migration every time we add a new setting though. 

What I've seen in the past, in code that isn't using your `ConfigurationMetadata` helpers, was a check done in the constructor to see if the setting wasn't set, and if so give the default value. 

For example:
- SIP2 
  - https://github.com/ThePalaceProject/circulation/blob/main/api/sip/client.py#L271
  - https://github.com/ThePalaceProject/circulation/blob/main/api/sip/client.py#L265-L266
- Millenium 
  - https://github.com/ThePalaceProject/circulation/blob/main/api/millenium_patron.py#L157-L158

Perhaps this could be handled by the `ConfigurationMetadata` code, so that is returns the default value for settings that are not in the database.

Reply:
Continuing the discussion from the weekly sync call this morning @vbessonov 

Here is the test I did:
- Check out commit 8ec48c7108d8453871d20c8d2a183c211681a62b (just before anything for LCP goes in)
- Create a new ODL2 collection.
- Update to the code in this branch.
- Load the CM admin UI screen for the ODL2 collection

This is what is displayed:
![Screen Shot 2021-11-08 at 1 03 06 PM](https://user-images.githubusercontent.com/569437/140786673-448beb7c-e059-4870-8f35-b591d4e3e2fb.png)

So the admin UI displays the default values for the values not in the database yet.